providers/idle_timeout.rb

```diff
@@ -0,0 +1,116 @@
+#
+# Cookbook Name:: gecos-ws-mgmt
+# Provider:: idle_timeout
+#
+# Copyright 2013, Junta de Andalucia
+# http://www.juntadeandalucia.es/
+#
+# All rights reserved - EUPL License V 1.1
+# http://www.osor.eu/eupl
+#
+
+
+
+action :setup do
+
+  begin
+
+    if new_resource.support_os.include?($gecos_os)
+
+      package "zenity" do
+        action :nothing
+      end.run_action(:install)
+     
+      package 'xautolock' do
+        action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+        gid = Etc.getpwnam(username).gid
+
+        Chef::Log.debug("idle_timeout ::: user = #{user}")
+        Chef::Log.debug("idle_timeout ::: username = #{username}")
+
+        autostart = ::File.expand_path("~#{username}/.config/autostart")
+        Chef::Log.debug("idle_timeout ::: autostart file = #{autostart}")
+
+        if user.idle_enabled
+
+          directory autostart do
+            owner username
+            group gid
+            recursive true
+            action :nothing
+            mode '0755'
+          end.run_action(:create)
+
+          cookbook_file "autolock.desktop" do
+            path "#{autostart}/autolock.desktop"
+            owner username
+            group gid
+            mode '0755'
+            action :nothing
+          end
+
+          template '/usr/bin/autolock.sh' do
+            source 'autolock.erb'
+            mode '0755'
+            variables ({
+              :timeout => user.idle_options['timeout'],
+              :command => user.idle_options['command'],
+              :notification => user.idle_options['notification']
+            })
+            notifies :create, 'cookbook_file[autolock.desktop]', :immediately
+            #only_if { ::File.executable?(user.idle_options['command']) }
+          end
+
+        else
+
+          %W(#{autostart}/autolock.desktop /usr/bin/autolock.sh).each do |f|
```

Comment:
These files will be erased if last user in policy does not have idle enabled

Reply:
Because of being user policy, it's been customized autolock.sh per user ($HOME/bin/autolock.sh)